interface/backend/api/views.py

```diff
@@ -55,3 +61,38 @@ def candidate_mark(request, candidate_id):
 
 def candidate_dismiss(request, candidate_id):
     return JsonResponse({'response': "Candidate {} was dismissed".format(candidate_id)})
+
+
+def generate_case_dict(case_id):
+    try:
+        case_dict = model_to_dict(Case.objects.get(pk=case_id))
+    except Exception as e:
+        return {'response': 'Case not found'}
+
+    series_id = case_dict['series']
+    case_dict['series'] = model_to_dict(ImageSeries.objects.get(pk=series_id))
+
+    candidates = [model_to_dict(candidate) for candidate in Candidate.objects.filter(case_id=case_id)]
+    for candidate in candidates:
+        candidate['centroid'] = model_to_dict(ImageLocation.objects.get(pk=candidate['centroid']))
+    case_dict['candidates'] = candidates
+
+    nodules = [model_to_dict(nodule) for nodule in Nodule.objects.filter(case_id=case_id)]
+    for nodule in nodules:
+        nodule['centroid'] = model_to_dict(ImageLocation.objects.get(pk=nodule['centroid']))
+    case_dict['nodules'] = nodules
+    return case_dict
+
+
+def json_serial(obj):
+    """JSON serializer for objects not serializable by default json code"""
+
+    if isinstance(obj, (datetime, date)):
+        return obj.isoformat()
+    raise TypeError("Type %s not serializable" % type(obj))
+
+
+def case_report(request, case_id):
+    case_dict = generate_case_dict(case_id)
+    case_pretty_json = json.dumps(case_dict, indent=4, sort_keys=True, default=json_serial)
+    return HttpResponse(case_pretty_json, content_type="application/json")
```

Comment:
Hm, do we not have a `JsonResponse` ?

Reply:
I'd also prefer pure JSON put the issue states `For now, this can be a simple view that takes a Case data, generates the corresponding dict structure and pretty-prints using the pprint module within a HTML page.` Doesn't that mean the rendered page should be a HTML page including the pretty printed JSON?

Reply:
In fact, sending `application/json` as content type makes it possible to use browser add-ons that make JSON explorable and even prettier:
![screenshot from 2017-09-10 19-27-55](https://user-images.githubusercontent.com/6676439/30251395-f3278932-965e-11e7-98ef-f1df093b598c.png)
Without such add-ons, the JSON is still printed in a nice way due to the HttpResponse.
![screenshot from 2017-09-10 19-28-09](https://user-images.githubusercontent.com/6676439/30251452-bc9aeaac-965f-11e7-9e73-1c992a370007.png)

Reply:
But surely JsonResponse sends exactly that? (Or we should have our own version?)

Reply:
When I use JsonResponse and try to pretty-print the JSON using `json.dumps(case_dict, indent=4, sort_keys=True..`, I end up with 
![screenshot from 2017-09-11 08-51-21](https://user-images.githubusercontent.com/6676439/30261768-66a1db0a-96ce-11e7-87bd-3775f76649e7.png)
which is obviously neither pretty nor valid JSON.
Without using `json.dumps` and without using a browser add-on for pretty JSON formatting, you end up with
![screenshot from 2017-09-11 08-49-56](https://user-images.githubusercontent.com/6676439/30261723-3722b4e4-96ce-11e7-8cc7-03e5f338acfa.png)
which is also not pretty.
So I'd clearly plead for the current HttpResponse and json.dumps implementation.

Reply:
@WGierke (cc @lamby) Seems to me neither `HttpResponse` nor `JsonResponse` is the idiomatic approach here.

Let's do this the DRF way, yeah? Please have a look at the DRF documentation, this is a solved problem: http://www.django-rest-framework.org/tutorial/2-requests-and-responses/

Reply:
Thanks @isms I wasn't aware of that idiom.
Now one can add a suffix that states what content type should be returned:
![screenshot from 2017-09-12 09-53-32](https://user-images.githubusercontent.com/6676439/30315528-92f8c08e-97a4-11e7-83c2-152c1264a1ac.png)
![screenshot from 2017-09-12 09-53-47](https://user-images.githubusercontent.com/6676439/30315529-92fee522-97a4-11e7-94dd-c1047e5cefa8.png)